sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessPipelineRunner.java

```diff
@@ -262,6 +264,11 @@ public InProcessPipelineResult run(Pipeline pipeline) {
     return result;
   }
 
+  private Map<Class<? extends PTransform>, Collection<ModelEnforcementFactory>>
+      defaultModelEnforcements(InProcessPipelineOptions options) {
+    return Collections.emptyMap();
```

Comment:
Yes. The option I would use for enforcing immutability requires additional work to ensure that outputs aren't mutated (done in [#86](https://github.com/apache/incubator-beam/pull/86))